bottle_app.py

```diff
@@ -629,14 +629,59 @@ def api_get_movie_trackpoints():
         labels         = sorted( ( tp['label'] for tp in trackpoint_dicts) )
         frame_dicts    = defaultdict(dict)
 
+        ruler0_x, ruler20_x = -1, -1
+        ruler0_y, ruler20_y = -1, -1
+        ruler0_frame_number = -1
+        sensor_width_mm = 36  # Sensor width 36 millimeters (this is a common value,
+        # but should be adjusted according to actual circumstances)
+        actual_distance_mm = 20
+        mm_per_inch = 25.4  # common value
+
         for tp in trackpoint_dicts:
-            frame_dicts[tp['frame_number']][tp['label']+' x'] = tp['x']
-            frame_dicts[tp['frame_number']][tp['label']+' y'] = tp['y']
+            frame_dicts[tp['frame_number']][tp['label'] + ' x'] = tp['x']
+            frame_dicts[tp['frame_number']][tp['label'] + ' y'] = tp['y']
+            if tp['label'] == 'ruller 0 mm':
+                ruler0_x = tp['x']
+                ruler0_y = tp['y']
+                ruler0_frame_number = tp['frame_number']
+            if tp['label'] == 'ruller 20 mm':
+                ruler20_x = tp['x']
+                ruler20_y = tp['y']
+
+                # calculate the dpi of the ruler, and convert to mm
+                def pixels_to_mm(pixel_value):
+                    pixel_distance_px = math.sqrt((ruler20_x - ruler0_x) ** 2 + (ruler20_y - ruler0_y) ** 2)
+                    pixel_to_mm_ratio = actual_distance_mm / pixel_distance_px
```

Comment:
It's not clear to me how it is possible to have a function that takes `pixel_value` and returns `pixels_to_mm`. Please recast this function (and move it to tracker.py).

Reply:
OK，I will do it.

Reply:
> It's not clear to me how it is possible to have a function that takes `pixel_value` and returns `pixels_to_mm`. Please recast this function (and move it to tracker.py).

Because, It's an inner function all the parameters should be share in the whole function, so I can only pass one parameter to calculate.

Reply:
Yes, but that makes it harder to test with a unit test.

Reply:
I will fix it.